packages/core/src/components/CodeInput/CodeInput.tsx

```diff
@@ -0,0 +1,94 @@
+import React from "react";
+import {
+  StyleProp,
+  ViewStyle,
+  TextInput as NativeTextInput,
+  View,
+} from "react-native";
+import TextInput, { TextInputProps } from "../TextInput";
+import {
+  CodeField,
+  useClearByFocusCell,
+} from "react-native-confirmation-code-field";
+import { DefaultCodeInputCell } from "./CodeInputCell";
+
+interface CellItem {
+  cellValue: string;
+  index: number;
+  isFocused: boolean;
+}
+
+interface CodeInputProps extends TextInputProps {
+  onInputFull?: (value: string) => void;
+  cellCount?: number;
+  clearOnCellFocus?: boolean;
+  blurOnFull?: boolean;
+  renderItem?: ({ cellValue, index, isFocused }: CellItem) => JSX.Element;
+  style?: StyleProp<ViewStyle>;
+}
+
+const CodeInput = React.forwardRef<NativeTextInput, CodeInputProps>(
+  (
+    {
+      onInputFull,
+      cellCount = 4,
+      clearOnCellFocus = true,
+      blurOnFull = true,
+      renderItem,
+      value,
+      onChangeText,
+      style,
+      ...rest
+    },
+    ref
+  ) => {
+    const newCodeInputRef = React.useRef<NativeTextInput>(null);
+
+    // Use the provided ref or default to new ref when not provided
+    const codeInputRef = ref
+      ? (ref as React.RefObject<NativeTextInput>)
+      : newCodeInputRef;
+
+    // Clears input of a cell when focused, configured as explained here (https://github.com/retyui/react-native-confirmation-code-field/blob/master/API.md#useclearbyfocuscellvalue-string-setvalue-text-string--void)
+    const [codeFieldProps, getCellOnLayout] = useClearByFocusCell({
+      value,
+      setValue: (text) => onChangeText?.(text),
+    });
+
+    React.useEffect(() => {
+      if (value?.length === cellCount) {
+        if (blurOnFull) {
+          codeInputRef.current?.blur();
+        }
+        onInputFull?.(value);
+      }
+      // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
In the case that it is an inline function like `onInputFull = {()=>{}}`, which is the most common case, then it would call this on every rerender due to the inline function being a new one every rerender. I don't see it as much of a problem, even it gets out of date, it will only be until the value (or any of the others) changes again